plaso/parsers/plist_plugins/macos_login_items.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+"""Plist parser plugin for MacOS login items plist files."""
+from datetime import datetime
+
+from dfdatetime import hfs_time as dfdatetime_hfs_time
+from plistutils.alias import AliasParser
+from plistutils.bookmark import BookmarkParser
+
+from plaso.containers import events
+from plaso.parsers import plist
+from plaso.parsers.plist_plugins import interface
+
+
+class MacOSLoginItemsEventData(events.EventData):
+  """MacOS login items event data.
+
+  Attributes:
+    name (str): name.
+    hidden (bool): whether this login item is hidden.
+    cnid_path: a "/" delimited string representing an array of CNIDs.
+    volume_name: the name of the volume containing the target.
+    target_path: the full filesystem path to the target.
+    volume_mount_point: the location the volume is mounted on the filesystem.
+    creation_date: creation date of the login item alias data.
+    volume_creation_date: creation date of the volume.
+    volume_flags: a string containing ", "-separated names of volume flags.
+  """
+
+  DATA_TYPE = 'macos:loginitems:entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(MacOSLoginItemsEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.name = None
+    self.hidden = None
+    self.cnid_path = None
+    self.volume_name = None
+    self.target_path = None
+    self.volume_mount_point = None
+    self.creation_date = None
+    self.volume_creation_date = None
+    self.volume_flags = None
+
+
+def _ConvertHFSDate(alias_date: datetime) -> dfdatetime_hfs_time.HFSTime:
+  hfs_epoch = dfdatetime_hfs_time.HFSTimeEpoch()
+  hfs_epoch_datetime = datetime(
+    year=hfs_epoch.year,
+    month=hfs_epoch.month,
+    day=hfs_epoch.day_of_month,
+  )
+  hfs_epoch_timedelta = alias_date - hfs_epoch_datetime
+  return dfdatetime_hfs_time.HFSTime(
+    timestamp=hfs_epoch_timedelta.total_seconds()
+  )
+
+
+def _ParseBookmark(
+  parser_mediator,
+  filename: str,
+  idx: int,
```

Comment:
Plaso does not use types yet, due incompatibility of different version of type annoations